### PR TITLE
Session timeout: Add request method, additional data, custom callback

### PR DESCRIPTION
**Pull request by bmadzar**

### CODE REVIEW

**Comment by LaurentGoderre** [comment]
Can you squash the commits and make the commit message the title of this PR?

**Comment by bmadzar** [comment]
Done!

**Comment by LaurentGoderre** [comment]
Wouldn't it be simple to keep the same call like this?

```
$.post( settings.refreshCallbackUrl, settings.additionalData, function( response ) {

}, "text" );
```

This should do the same thing while staying shorter

**Comment by andre-gh** [comment]
It might be cleaner to set `context` to `$elm` and use `.done(keepaliveresponse)` and move the callback out of the function scope using `this` to access the `$elm`

**Comment by bmadzar** [comment]
The main motivation for this pull request was to allow the request method used to be set as a configuration option. We currently have to apply a workaround on the server side to be able to handle this request using POST. This is the reason for `$.ajax` instead of `$.post`.

**Comment by LaurentGoderre** [comment]
@bmadzar I understand. WHat I am saying is that you can pass data by simply modifying the existing $.post() call.

What is the use case to return results to the plugin? I doubt you could do anything with it without modifying the plugin.

**Comment by bmadzar** [comment]
@LaurentGoderre What I meant was that it would be nice if the response checking was customizable; i.e. a custom callback function that would accept the response text from the server and return true or false to the plugin. The requirement of the server having to return the literal value "true" is a bit rigid.

We've faced three minor issues at various times while using this plugin:
1. POST is required, whereas for this type of request we would use GET (fixed in this PR)
2. No ability to send additional data to the server (fixed in this PR)
3. No ability to implement custom response checking (not sure how best to do this)

Hope that helps clarify the motivation a bit.

**Comment by LaurentGoderre** [comment]
AAh so you want the ability to sometime use GET? If so that would make more sense.

**Comment by LaurentGoderre** [comment]
You could move

```js
if ( response && response.replace( /\s/g, "" ) === "true" ) {
	$elm.trigger( resetEvent, settings );
}
```
 to a function called isSessionValid like such

```js
settings: {
	...
	isSessionValid: function( response ) {
		return response && response.replace( /\s/g, "" ) === "true"
	}
}
```

This way, you can override it with your own function. You would then change the callback to:

```js
// Session is valid
if ( settings.isSessionValid( response ) ) {
	$elm.trigger( resetEvent, settings );

// Session has timed out - let the user know they need to sign in again
} else {
...
}
```

**Comment by bmadzar** [comment]
@LaurentGoderre I've updated the patch, adding an optional custom callback.

File: src/plugins/session-timeout/session-timeout.js [hunk 1]
```diff
@@ -36,7 +36,13 @@ var $modal, $modalLink, countdownInterval, i18n, i18nText,
 		refreshCallbackUrl: null,	/* refresh callback if using AJAX keepalive (no default) */
 		logouturl: "./",			/* logout URL once the session has expired */
 		refreshOnClick: true,		/* refresh session if user clicks on the page */
-		refreshLimit: 200000		/* default period of 2 minutes (ajax calls happen only once during this period) */
+		refreshLimit: 200000,		/* default period of 2 minutes (ajax calls happen only once during this period) */
+		method: "POST",				/* the request method to use */
+		additionalData: null,		/* additional data to send with the request */
+		sessionaliveCallback:		/* callback function used to check the server response */
+		function( response ) {
```
Comment (LaurentGoderre) [inline src/plugins/session-timeout/session-timeout.js]: Can you make this be on the previous line and indent the content of the function

```
sessionaliveCallback: function( response ) {		/* callback function used to check the server response */
	return response.replace( /\s/g, "" ) === "true";
}
+		
```

**Comment by bmadzar** [comment]
Fixed spacing

**Comment by LaurentGoderre** [comment]
Good to go on my end

**Comment by LaurentGoderre** [comment]
Oooh, actually, could you document the callback in your PR?

**Comment by bmadzar** [comment]
Updated documentation, example page

**Comment by LaurentGoderre** [comment]
This is good to go on my end. @masterbee @pjackson28 any issue with this?

**Comment by pjackson28** [comment]
Looks good to me. Have all the normal use cases been tested to ensure they aren't breaking? Just want to make sure that this won't negatively impact existing implementations.

**Comment by LaurentGoderre** [comment]
The existing unit tests all pass. So as far as we know it won't break anything.

**Pull request merged**